openfl/component/assigner/assigner.py

```diff
@@ -93,3 +107,50 @@ def get_aggregation_type_for_task(self, task_name):
         if "aggregation_type" not in self.tasks[task_name]:
             return None
         return self.tasks[task_name]["aggregation_type"]
+
+    @classmethod
+    def with_selected_task_group(cls, func):
+        """Decorator to filter task groups based on selected_task_group.
+
+        This decorator should be applied to define_task_assignments() method
+        in Assigner subclasses to handle task_group filtering.
+        """
+
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            # Check if selection of task_group is applicable
+            if hasattr(self, "selected_task_group") and self.selected_task_group is not None:
+                # Verify task_groups exists before attempting filtering
+                if not hasattr(self, "task_groups"):
+                    logger.warning(
+                        "Task group specified for selection but no task_groups found. "
+                        "Skipping filtering. This might be intentional for custom assigners."
+                    )
+                    return func(self, *args, **kwargs)
+
+                assert self.task_groups, "No task_groups defined in assigner."
+
+                # Perform the filtering
+                selected_task_groups = [
+                    group for group in self.task_groups if group["name"] == self.selected_task_group
```

Comment:
What happens if someone accidentally / purposefully lists "learning" or "evaluation" twice?

Reply:
Nevermind! I see this is handled